scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -0,0 +1,1226 @@
+        @param: starting - a list of DockerLoggers to start
+        """
+        if starting:
+            self._logger.log(scalyr_logging.DEBUG_LEVEL_2, 'Starting all docker loggers')
+            docker_logs = self.__get_docker_logs( starting, k8s_data )
+            self.__start_docker_logs( docker_logs )
+            self.docker_logs.extend( docker_logs )
+
+    def __start_docker_logs( self, docker_logs ):
+        for log in docker_logs:
+            if self.__log_watcher:
+                log['log_config'] = self.__log_watcher.add_log_config( self.__module, log['log_config'] )
+
+            self.raw_logs.append( log )
+
+    def __get_last_request_for_log( self, path ):
+        result = datetime.datetime.fromtimestamp( self.__start_time )
+
+        try:
+            full_path = os.path.join( self.__log_path, path )
+            fp = open( full_path, 'r', self.__readback_buffer_size )
+
+            # seek readback buffer bytes from the end of the file
+            fp.seek( 0, os.SEEK_END )
+            size = fp.tell()
+            if size < self.__readback_buffer_size:
+                fp.seek( 0, os.SEEK_SET )
+            else:
+                fp.seek( size - self.__readback_buffer_size, os.SEEK_SET )
+
+            first = True
+            for line in fp:
+                # ignore the first line because it likely started somewhere randomly
+                # in the line
+                if first:
+                    first = False
+                    continue
+
+                dt, _ = _split_datetime_from_line( line )
+                if dt:
+                    result = dt
+            fp.close()
+        except Exception, e:
+            global_log.info( "%s", str(e) )
+
+        return scalyr_util.seconds_since_epoch( result )
+
+    def __create_log_config( self, parser, path, attributes, parse_as_json=False ):
+        """Convenience function to create a log_config dict from the parameters"""
+
+        return { 'parser': parser,
+                 'path': path,
+                 'parse_lines_as_json' : parse_as_json,
+                 'attributes': attributes
+               }
+
+    def __get_base_attributes( self ):
+        attributes = None
+        try:
+            attributes = JsonObject( { "monitor": "agentKubernetes" } )
+            if self.__host_hostname:
+                attributes['serverHost'] = self.__host_hostname
+
+        except Exception, e:
+            self._logger.error( "Error setting monitor attribute in KubernetesMonitor" )
+            raise
+
+        return attributes
+
+    def __get_log_config_for_container( self, cid, info, k8s_data, base_attributes ):
+        result = None
+
+        container_attributes = base_attributes.copy()
+        container_attributes['containerName'] = info['name']
+        container_attributes['containerId'] = cid
+
+        parser = 'docker'
+        if cid in k8s_data:
+            pod = k8s_data[cid]
+
+            container_attributes['podName'] = pod.name
```

Comment:
Change `podName` to `pod_name` unless if there is a really strong convenient of using camel case for k8s.